src/pudl/package_data/notebooks/how_to_access_pudl.ipynb

```diff
@@ -0,0 +1,222 @@
+{
```

Comment:
<p>This is kind of a special case -- constructing a whole-table <code>SELECT * FROM table</code> statement, and then passing it to Pandas, rather than using SQLAlchemy's query building facilities. What functionality are we trying to demonstrate here? The fact that you can pass an SQLAlchemy select to <code>df.read_sql()</code> instead of a raw SQL string? Or are we trying to highlight how one might build up a query using SQLAlchemy's interface rather than raw SQL?</p><p> </p><p>To me a natural progression seems like it might be:</p><ul><li>We've given you an SQLite database, and you can access it like any other SQLite database e.g. via Python's built-in <a href="https://www.sqlitetutorial.net/sqlite-python/" target="_blank">sqlite3</a> library, or the more extensive <a href="https://realpython.com/python-sqlite-sqlalchemy/" target="_blank">SQLAlchemy tools</a>.</li><li>However, for interactive analysis, you'll probably be using Pandas, which <a href="https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_sql.html" target="_blank">provides some nice shortcuts</a> for grabbing data from databases as dataframes, including grabbing a whole table by name, or returning the results of a query.</li><li>Here's how you get a list of all the tables in the database, and all the columns in a table.</li><li>If you want to pass Pandas a query, it can either be raw SQL, or you can use an SQLAlchemy query object.</li></ul><p> </p><p>And if we're going to give an example of using a query instead of just grabbing a whole table, I think it might be more informative if it's a (simple) query that does something other than just grabbing a whole table, which is kind of a special case that may not make it clear what's happening, and that it's different from the other "gimme a table" cases.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/catalyst-cooperative/pudl/pull/844/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>